### PR TITLE
Course awards component

### PR DESCRIPTION
**Pull request by quinnbudan**
Screenshot of the component:
course_awards_no_wrap

Screenshot of the add awards popup subcomponent:
add_award_popup

### CODE REVIEW

File: src/components/add-awards-dialog.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+		const action = await d2lDialog.open();
+
+		if (action === DONE_ACTION) {
+			const d2lList = d2lDialog.querySelector('d2l-list');
+			const { keys } = d2lList.getSelectionInfo();
+			const awards = this.availableAwards.filter(award => keys.includes(award.id));
+
+			return awards;
+		}
+		return [];
```
Comment (BraydonKains) [inline src/components/add-awards-dialog.js:54]: We might want to decide on a better failure state for the dialogs and do it on both. This was part of the reason that I was thinking the actual `d2l-dialog` would make more sense outside of this component.

**Comment by louiscoste** [comment]
Looking at the UI I do not think I like how the `Hidden until earned` column looks. Maybe we can talk later about this

**Comment by quinnbudan** [comment]
> Looking at the UI I do not think I like how the `Hidden until earned` column looks. Maybe we can talk later about this

I agree with you, and I think I've addressed it. Please take another look!

File: src/components/course-awards.js [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+		const { target: { value: q } } = event;
+		console.log(q);
+
+		// replace with API call
+		this.courseAwards.forEach(award =>
+			award.doShow = q === 'All Awards' || award.type.toUpperCase() === q.toUpperCase().substring(0, q.length - 1));
+		this.requestUpdate();
+	}
+
+	_renderInitialComponents() {
```
Comment (BraydonKains) [inline src/components/course-awards.js:128]: "header" may be a better name here, we can standardize that too

File: src/components/add-awards-dialog.js (hunk 1 above)
Comment (quinnbudan) [reply src/components/add-awards-dialog.js:54]: But is it a failure state to not select any awards to add?

**Pull request merged**